ibm/service/kubernetes/resource_ibm_container_dedicated_host_pool.go

```diff
@@ -4,23 +4,35 @@
 package kubernetes
 
 import (
+	"context"
 	"fmt"
+	"log"
+	"time"
 
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 
 	v2 "github.com/IBM-Cloud/bluemix-go/api/container/containerv2"
 	"github.com/IBM-Cloud/bluemix-go/bmxerror"
 	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
 )
 
+const (
+	DedicatedHostPoolStateCreated  = created
+	DedicatedHostPoolStateCreating = creating
+	DedicatedHostPoolStateDeleting = deleting
+	DedicatedHostPoolStateDeleted  = deleted
+)
+
 func ResourceIBMContainerDedicatedHostPool() *schema.Resource {
 
 	return &schema.Resource{
-		Create:   resourceIBMContainerDedicatedHostPoolCreate,
-		Read:     resourceIBMContainerDedicatedHostPoolRead,
-		Delete:   resourceIBMContainerDedicatedHostPoolDelete,
-		Importer: &schema.ResourceImporter{},
-		Timeouts: &schema.ResourceTimeout{},
+		CreateContext: resourceIBMContainerDedicatedHostPoolCreate,
+		ReadContext:   resourceIBMContainerDedicatedHostPoolRead,
+		DeleteContext: resourceIBMContainerDedicatedHostPoolDelete,
+		Importer:      &schema.ResourceImporter{},
+		Timeouts:      &schema.ResourceTimeout{},
```

Comment:
Define some default timeouts for all create,update,delete

Reply:
done